src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -101,43 +109,92 @@ internal CloudTable GetDiagnosticEventsTable(DateTime? now = null)
             return _diagnosticEventsTable;
         }
 
-        private static string GetCurrentTableName(DateTime now)
+        private static string GetTableName(DateTime date)
         {
-            return $"{TableNamePrefix}{now:yyyyMM}";
+            return $"{TableNamePrefix}{date:yyyyMM}";
         }
 
         protected internal virtual async void OnFlushLogs(object state)
         {
             await FlushLogs();
         }
 
+        private async Task PurgePreviousEventVersions()
+        {
+            _logger.LogTrace("Purging previous diagnostic event versions.");
+
+            try
+            {
+                var tables = await TableStorageHelpers.ListTablesAsync(TableClient, TableNamePrefix);
+
+                foreach (var table in tables)
+                {
+                    var tableRecords = await table.ExecuteQuerySegmentedAsync(new TableQuery<DiagnosticEvent>(), null);
+
+                    // Skip tables that have 0 records
+                    if (tableRecords.Results.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    // Delete table if it doesn't have records with EventVersion
+                    var eventVersionDoesNotExists = tableRecords.Results.Any(record => string.IsNullOrEmpty(record.EventVersion) == true);
+                    if (eventVersionDoesNotExists)
+                    {
+                        await table.DeleteIfExistsAsync();
+                        continue;
+                    }
+
+                    // If the table does have EventVersion, query if it is an outdated version
+                    var eventVersionOutdated = tableRecords.Results.Any(record => record.EventVersion.CompareTo(DiagnosticEvent.CurrentEventVersion) < 0);
+                    if (eventVersionOutdated)
+                    {
+                        await table.DeleteIfExistsAsync();
+                    }
+                }
+
+                _purged = true;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred when attempting to purge previous diagnostic event versions.");
+            }
+        }
+
         internal virtual async Task FlushLogs(CloudTable table = null)
         {
             if (_environment.IsPlaceholderModeEnabled())
             {
                 return;
             }
 
+            if (_primaryHostStateProvider.IsPrimary && !_purged)
+            {
+                // await PurgePreviousEventVersions();
+                var tIgnore = Task.Run(async () => { await PurgePreviousEventVersions(); });
```

Comment:
To ignore or not to ignore, that is the question

Reply:
If ignoring, you can use a [discard](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards) here. 

You should also be able to call `PurgePreviousEventVersions` without awaiting and directly assigning to a discard.

Reply:
@brettsam updated this to purge in background (`_ = PurgePreviousEventVersions();` - if you feel strongly about otherwise feel free to change.